deltachat-ios/AppDelegate.swift

```diff
@@ -447,38 +462,46 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
             }
         }
 
-        // we're in background, run IO for a little time
-        dcAccounts.startIo()
-        dcAccounts.maybeNetwork()
+        pushToDebugArray("s1")
+
+        DispatchQueue.global(/*.default qos as .background is not always executed*/).async { [weak self] in
 
-        DispatchQueue.main.asyncAfter(deadline: .now() + 10) { [weak self] in
-            logger.info("⬅️ finishing fetch")
             guard let self = self else {
                 completionHandler(.failed)
                 return
             }
 
+            // we're in background, run IO for a little time
+            self.dcAccounts.startIo()
+            self.pushToDebugArray("s2")
+            self.dcAccounts.maybeNetwork()
+            self.pushToDebugArray("s3")
+
+            _ = fetchSemaphore.wait(timeout: .now() + 10)
```

Comment:
this is still where the app sometimes hangs and the semaphore never returns.

might be ios takes the waking as a chance to freeze the app again. _much_ too late, the app returns then, without getting messages in time.

this does not happen often, but it happens. maybe w can try if a look with some sleep() works better.